scripts/reboot

```diff
@@ -215,6 +350,55 @@ function linecard_reboot_notify_supervisor()
     fi
 }
 
+# Function to reboot all DPUs in parallel
+function reboot_all_dpus() {
+    local NUM_DPU=$1
+
+    for (( i=0; i<"$NUM_DPU"; i++ )); do
+        echo "Rebooting DPU module dpu$i"
+        reboot_dpu_module "dpu$i" &
+    done
+    wait
+}
+
+# Function to handle scenarios on smart switch
+function handle_smart_switch() {
+    if [ -f "$PLATFORM_JSON_PATH" ]; then
+        NUM_DPU=$(jq -r '.DPUS | length' "$PLATFORM_JSON_PATH" 2>/dev/null)
+        if [ "$NUM_DPU" -gt 0 ]; then
+            SMART_SWITCH="yes"
+        fi
+    fi
+
+    if [[ "$REBOOT_DPU" == "yes" ]]; then
+        if [[ "$SMART_SWITCH" == "yes" ]]; then
+            echo "User requested to reboot the device ${DPU_MODULE_NAME}"
+            reboot_dpu_module "$DPU_MODULE_NAME"
+        else
+            echo "Invalid '-d' option specified for a non-smart switch"
+            exit ${EXIT_ERROR}
+        fi
+    fi
+
+    if [[ "$SMART_SWITCH" == "yes" ]]; then
+        is_dpu=$(python3 -c "import reboot_helper; reboot_helper.is_dpu()")
+
+        # Check if system is a DPU and handle -p option accordingly
+        if [[ "$is_dpu" == "True" && "$PRE_SHUTDOWN" != "yes" ]]; then
+            echo "Invalid, '-p' option not specified for a DPU"
+            exit ${EXIT_ERROR}
+        elif [[ "$is_dpu" != "True" && "$PRE_SHUTDOWN" == "yes" ]]; then
+            echo "Invalid '-p' option specified for a non-DPU"
+            exit ${EXIT_ERROR}
+        fi
+
+        # If not a DPU, reboot all DPUs in parallel
+        if [[ "$is_dpu" != "True" ]]; then
+            reboot_all_dpus "$NUM_DPU"
```

Comment:
If NPUs + DPUs need reboot we need to pass the `MODULE_REBOOT_SMARTSWITCH` to the module reboot function, if individual DPUs need reboot we use the `MODULE_REBOOT_DPU` as per reboot hld:

reboot(self, reboot_type):
```
Define new reboot_type as MODULE_REBOOT_DPU for DPU only reboot and MODULE_REBOOT_SMARTSWITCH for entire switch reboot.

```

Reply:
I will fix the HLD, it was not updated. Thank you!

Reply:
The reboot script should execute the reboot of DPUS in parallel. I think the differentiation between `MODULE_REBOOT_DPU` and `MODULE_REBOOT_SMARTSWITCH` is used by platforms to decide whether to wait for the dpus to finish rebooting or not (if we are rebooting individual DPUs -> `MODULE_REBOOT_DPU` -> wait for DPU to finish rebooting. If we are rebooting all DPUS + switch -> `MODULE_REBOOT_SMARTSWITCH`-> NPU will just issue reboot command to DPUs and not wait for the result as it is undergoing reboot itself ) Correct me if I am wrong

Reply:
@vvolam we agreed that `MODULE_REBOOT_SMARTSWITCH` would be used to initiate DPU reboots during a system reboot, without waiting for the DPUs to fully boot, to reduce overall reboot time. What has changed? Why is there a need to update the HLD?